### ISSUE TITLE
How to approach add_nested with mismatched index

### ISSUE THREAD

**Issue by dougbrn** [lincc-frameworks/nested-pandas#42]
add_nested currently works as a left join from the perspective of the base layer, as shown below:
```
# unmatched nested objects are removed
>>> base = NestedFrame({"a":[1,2,3]}, index=[0,1,2])
>>> nested = NestedFrame({"b":[1,2,3,4]}, index=[0,1,2,3])

>>> base.add_nested(nested, "nested")
	a	nested
0	1	b 0 1
1	2	b 0 2
2	3	b 0 3
```

```
# unmatched base rows are kept
>>> base = NestedFrame({"a":[1,2,3]}, index=[0,1,2])
>>> nested = NestedFrame({"b":[1,2,3,4]}, index=[0,1,3,3]) # no match for index=2

>>> base.add_nested(nested, "nested")
	a	nested
0	1	b 0 1
1	2	b 0 2
2	3	None
```

Is this the best default behavior? Should we add various join options to the output of this operation? For time-domain purposes, I can see value in keeping nested lightcurves even when no object information is present

**Comment by dougbrn** [comment]
This has been implemented on the nested-dask side as a choice that defaults to an outer join, we should follow suit here:
<img width="818" alt="Screen Shot 2024-06-26 at 2 49 03 PM" src="https://github.com/lincc-frameworks/nested-pandas/assets/8853505/ad274455-1f02-49d5-b0bc-ec21d82b7b83">

**Comment by hombit** [comment]
Should be fixed when #115 and #116 are fixed
